### PR TITLE
Proper multiplication overflow test

### PR DESCRIPTION
**Pull request by eagleoflqj**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/radareorg/radare2/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/radareorg/radare2/blob/master/DEVELOPERS.md#code-style)
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the documentation and the [radare2 book](https://github.com/radareorg/radare2book) with the relevant information (if needed)

**Detailed description**

Even `a * 4 >= a`, the multiplication may still overflow.
Eg: `a = 0x58000000, a * 4 = 0x60000000`
Instead, we should use division to test whether a*b overflows.

**Test plan**

...

**Closing issues**

...

### CODE REVIEW

**Comment by karliss** [comment]
Instead of doing potentially overflowing multiplication which may be undefined behavior depending n argument type and trying to check after the fact. It is possible to do
```c
if (n > INT32_MAX / sizeof(foo)) {
    error;
}
amount = n * sizeof(foo);
```

Since INT32_MAX and  sizeof is compile time constant as additional bonus division gets avoided.

File: libr/bin/p/bin_bflt.c [hunk 1]
```diff
@@ -181,22 +180,17 @@ static RList *relocs(RBinFile *bf) {
...
 		if (!reloc_table) {
 			goto out_error;
 		}
 		amount = n_reloc * sizeof (ut32);
-		if (amount < n_reloc || amount > UT32_MAX) {
-			free (reloc_table);
-			goto out_error;
-		}
-		ut32 *reloc_pointer_table = calloc (1, amount + 1);
+		ut32 *reloc_pointer_table = calloc (1, amount);
```
Comment (radare) [inline libr/bin/p/bin_bflt.c:193]: amount can be 0, so this will be UB

File: libr/bin/p/bin_bflt.c [hunk 2]
```diff
@@ -144,20 +144,19 @@ static int get_ngot_entries(struct r_bin_bflt_obj *obj) {
...
 		if (n_got) {
-			amount = n_got * sizeof (ut32);
-			if (amount < n_got || amount > UT32_MAX) {
+			if (n_got > UT32_MAX / sizeof (struct reloc_struct_t)) {
 				goto out_error;
 			}
-			struct reloc_struct_t *got_table = calloc (
-				1, n_got * sizeof (struct reloc_struct_t));
+			amount = n_got * sizeof (struct reloc_struct_t);
+			struct reloc_struct_t *got_table = calloc (1, amount);
```
Comment (radare) [inline libr/bin/p/bin_bflt.c:159]: same

Comment (radare) [reply libr/bin/p/bin_bflt.c:159]: just noticed this after the 2 aprovals. can you have a look at it?

File: libr/bin/p/bin_bflt.c (hunk 1 above)
Comment (eagleoflqj) [reply libr/bin/p/bin_bflt.c:193]:
```c
if (obj->hdr->reloc_count > 0) {
		ut32 n_reloc = obj->hdr->reloc_count;
		if (n_reloc > UT32_MAX / sizeof (struct reloc_struct_t)) {
			goto out_error;
		}
		amount = n_reloc * sizeof (struct reloc_struct_t);
```
Here n_reloc>=1, so amount can't be zero.

File: libr/bin/p/bin_bflt.c (hunk 2 above)
Comment (eagleoflqj) [reply libr/bin/p/bin_bflt.c:159]: `if(ngot)`
guaranteed ngot>=1 and amount >=1

**Pull request merged**